python/ray/monitor.py

```diff
@@ -111,10 +111,14 @@ def __init__(self, redis_address, redis_port, autoscaling_config):
         self.issue_gcs_flushes = "RAY_USE_NEW_GCS" in os.environ
         self.gcs_flush_policy = None
         if self.issue_gcs_flushes:
-            # For now, we take the primary redis server to issue flushes,
-            # because task table entries are stored there under this flag.
+            # Data is stored under the first data shard, so we issue flushes to
+            # that redis server.
+            addr_port = self.redis.lrange("RedisShards", 0, -1)[0]
+            addr_port = addr_port.split(b":")
+            self.redis_shard = redis.StrictRedis(
+                host=addr_port[0], port=addr_port[1])
             try:
-                self.redis.execute_command("HEAD.FLUSH 0")
+                self.redis_shard.execute_command("HEAD.FLUSH 0")
```

Comment:
Why was this change necessary? Is `self.redis` pointing to the wrong server?

Also, is this code run every time a flush is required? If so, we should try to avoid running the `self.redis.lrange` command every time.